src/html/html-style-scanner.ts

```diff
@@ -54,15 +54,16 @@ export class HtmlStyleScanner implements HtmlScanner {
               true));
         } else {
           const contents = dom5.getTextContent(node);
-          const locationOffset = getLocationOffsetOfStartOfTextContent(node);
+          const locationOffset =
+              getLocationOffsetOfStartOfTextContent(node, document);
           const commentText = getAttachedCommentText(node) || '';
           features.push(new ScannedInlineDocument(
               'css',
               contents,
               locationOffset,
               commentText,
               document.sourceRangeForNode(node)!,
-              node));
+              {language: 'html', node}));
```

Comment:
Generally we try not to have up references like that, don't we? 

I do think features more generally having a pointer to their Document and ParsedDocument would often be pretty useful though.

In any case, this turned out to be quite useful in stringify, so I added `containingDocument` here. I kept `language` because TypeScript doesn't do type narrowing as well as we'd like:

https://www.typescriptlang.org/play/#src=interface%20Doc%3CNode%3E%20%7B%0D%0A%20%20%20%20readonly%20body%3A%20ReadonlyArray%3CNode%3E%3B%0D%0A%7D%0D%0Ainterface%20NodeAndDoc%3CNode%3E%20%7B%0D%0A%20%20%20%20readonly%20node%3A%20Node%3B%0D%0A%20%20%20%20readonly%20doc%3A%20Doc%3CNode%3E%3B%0D%0A%7D%3B%0D%0A%0D%0Adeclare%20const%20container%3A%20NodeAndDoc%3Cstring%7Cnumber%3E%3B%0D%0A%0D%0Aif%20(typeof%20container.node%20%3D%3D%3D%20'string')%20%7B%0D%0A%20%20%20%20const%20node%20%3D%20container.node%3B%0D%0A%20%20%20%20const%20firstChild%20%3D%20container.doc.body%5B0%5D%3B%0D%0A%20%20%20%20%2F%2F%20node%20and%20firstChild%20must%20both%20be%20strings%2C%20but%20typescript%0D%0A%20%20%20%20%2F%2F%20still%20thinks%20firstChild%20could%20be%20a%20number.%0D%0A%7D%0D%0A